### PR TITLE
Repository

### PR DESCRIPTION
**Pull request by IamHaddad**
faghat data ta viewModel miad, baraye review ba ejaze pr midam

### CODE REVIEW

File: app/src/main/java/com/worldsnas/starplayer/api/MusicResponse.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.worldsnas.starplayer.api
+
+data class MusicResponse(
+    var id: Long,
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/api/MusicResponse.kt]: loftan hatman annotation haro bezanid
albate man pishnahad mikonam az `Generating adapter` haye moshi estefade konin

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.worldsnas.starplayer.api
+
+import retrofit2.http.GET
+
+interface WebServiceApi {
+
+    @GET("/api/request")
+   suspend fun getMusics(): List<MusicResponse>
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt]: shakhsan dost nadaram `Response<>` ro hazf konam
etellaate mofide ziadi tosh has
hala har tor khodeton dost darid

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.worldsnas.starplayer.api
+
+import retrofit2.http.GET
+
+interface WebServiceApi {
+
+    @GET("/api/request")
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt]: chera hich parametri pas nemidim?
server bedone meghad chi bar migardone? (hame ye list ro?!!!!!)

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.worldsnas.starplayer.api
+
+import retrofit2.http.GET
+
+interface WebServiceApi {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt:7]: momkene API haye dige ham dashte bashim
esme monaseb tar bezarid behtare

File: app/src/main/java/com/worldsnas/starplayer/di/modules/NetworkModule.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        httpLoggingInterceptor: HttpLoggingInterceptor
+    ): OkHttpClient {
+        return OkHttpClient.Builder()
+            .addInterceptor(httpLoggingInterceptor)
+            .build()
+    }
+
+    @Provides
+    @JvmStatic
+    @Singleton
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/di/modules/NetworkModule.kt]: niazi nist inja be in `scope`

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusic.kt [hunk 6]
```diff
@@ -1,14 +1,10 @@
 package com.worldsnas.starplayer.model
 
-import android.os.Parcelable
-import kotlinx.android.parcel.Parcelize
-
-@Parcelize
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/LocalMusic.kt:6]: chera haz kardid ino?!
in model baraye ertebate 2 fragment bod faghat
age mikhayn esmesho ro behtar konin vali bayad Parcelable bashe

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProvider.kt [hunk 7]
```diff
@@ -2,5 +2,5 @@ package com.worldsnas.starplayer.model
 
 interface LocalMusicProvider {
 
-    suspend fun getAllMusic() : List<Music>
+    suspend fun getAllMusic() : List<LocalMusic>
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProvider.kt]: in dorost nist
ma object aslimon hamon `Music` hastesh va local music estefade dige i dare

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt [hunk 8]
```diff
@@ -21,12 +20,12 @@ constructor(private val contentResolver: ContentResolver) : LocalMusicProvider {
     )
     private val uri: Uri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI
 
-    override suspend fun getAllMusic(): List<Music> {
+    override suspend fun getAllMusic(): List<LocalMusic> {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt]: hamchenin

File: app/src/main/java/com/worldsnas/starplayer/model/Music.kt [hunk 9]
```diff
@@ -1,10 +1,14 @@
 package com.worldsnas.starplayer.model
 
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/Music.kt:6]: iin laye payini hast. nabayad az android khabar dashte bashe

File: app/src/main/java/com/worldsnas/starplayer/model/Repository.kt [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package com.worldsnas.starplayer.model
+
+import com.worldsnas.starplayer.api.MusicResponse
+
+interface Repository {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/Repository.kt]: esme repo ro mishe behtar kard alan

File: app/src/main/java/com/worldsnas/starplayer/model/Repository.kt [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+
+    suspend fun getApiData(): List<MusicResponse>
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/Repository.kt]: type e khoroji repo bayad az jensi bashe ke gharar dad kardim `Music`

File: app/src/main/java/com/worldsnas/starplayer/model/Repository.kt [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+    suspend fun getLocalData(): List<LocalMusic>
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/Repository.kt]: in ham hamchenin
vazife repo hastesh ke inaro tabdil kone

File: app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    private val localMusicProvider: LocalMusicProvider,
+    private val webServiceApi: WebServiceApi
+) : Repository {
+
+    override suspend fun getApiData(): List<MusicResponse> {
+
+        return withContext(Dispatchers.IO) {
+            val musicList = webServiceApi.getMusics()
+            Log.d("myTag", musicList.toString())
+            return@withContext musicList
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt]: error ha pas chi?
age null bod chi?
va .....

File: app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+import javax.inject.Inject
+
+class RepositoryImpl @Inject constructor(
+    private val localMusicProvider: LocalMusicProvider,
+    private val webServiceApi: WebServiceApi
+) : Repository {
+
+    override suspend fun getApiData(): List<MusicResponse> {
+
+        return withContext(Dispatchers.IO) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt]: fek konam khode retrofit mibare ro `IO` va niazi nabashe ma bezanim
search konid age intor hast ino hazf konim

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 15]
```diff
@@ -5,29 +5,31 @@ import androidx.lifecycle.LiveData
...
 import javax.inject.Inject
 
-class MusicListViewModel @Inject constructor(private val localMusicProvider: LocalMusicProvider) :
+class MusicListViewModel @Inject constructor(
+    private val repositoryImpl: RepositoryImpl
+) :
     ViewModel() {
     init {
-        getAllMusic()
+        getLocalMusic()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt:19]: esme function va on `object` fek konam baes shod in eshtebah etefagh biofte!

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 16]
```diff
@@ -5,29 +5,31 @@ import androidx.lifecycle.LiveData
...
 
-    private fun getAllMusic() {
+    private fun getLocalMusic() {
         viewModelScope.launch {
-            val musicsList = localMusicProvider.getAllMusic()
+            val musicsList = repositoryImpl.getApiData()
+            Log.d("myTag", musicsList.toString())
 
-            postMusicList.postValue(musicsList)
+//            postMusicList.postValue(musicsList)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: chera?
alan app dige hichi neshon nemide?!
chera feature haye dige ro darid dast mizanid?

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 17]
```diff
@@ -6,11 +6,11 @@ import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import com.worldsnas.starplayer.core.BaseViewHolder
 import com.worldsnas.starplayer.databinding.ItemMusicBinding
-import com.worldsnas.starplayer.model.LocalMusic
 import com.worldsnas.starplayer.model.Music
+import com.worldsnas.starplayer.model.LocalMusic
 
-class MusicsListAdapter(private var onItemClick: (LocalMusic) -> Unit) :
-    ListAdapter<Music, MusicListItemViewHolder>(DiffCallback) {
+class MusicsListAdapter(private var onItemClick: (Music) -> Unit) :
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt:12]: in ham dorost konid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 18]
```diff
@@ -64,7 +64,7 @@ class MusicsListFragment : Fragment() {
 
     private fun liveDataSetup() {
 
-        val musicObserver = Observer<List<Music>> {
+        val musicObserver = Observer<List<LocalMusic>> {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: hamchenin

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
good start 
let's see the improvements now

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt (hunk 16 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: baraye inke bbinam dorost kar mikone, felan jayegozin kardam

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusic.kt (hunk 6 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusic.kt:6]: daqiqan be dalile taqire nam

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProvider.kt (hunk 7 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProvider.kt]: hame tabdil ha ro dakhel Repository anjam bedam, doroste?

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt (hunk 8 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt]: na dige, ta Repo Local Music biaram baad tabdil konm be Music

File: app/src/main/java/com/worldsnas/starplayer/model/Music.kt (hunk 9 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/Music.kt:6]: mn yekam gij shodam, alan Music bere be laye haye balatar dige!
Response va LocalMusic har 2 miad dakhele repo, Music mire be ViewModel, in nazare mane(be in shekl taqir bedam)

File: app/src/main/java/com/worldsnas/starplayer/model/Repository.kt (hunk 10 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/Repository.kt]: MusicRepository khobe?

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProvider.kt (hunk 7 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProvider.kt]: hefze object `Music` kheili moheme
va bale tabdil ha bayad onja anjam beshe
baraye sample mitonid repo haye [in project mano](https://github.com/worldsnas/aio) ro bebinid

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt (hunk 8 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt]: `LocalMusic` vazifash chize dige ie
in vazayef ro ghati nakonim zendegi shirin khahad shod!

File: app/src/main/java/com/worldsnas/starplayer/model/Music.kt (hunk 9 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/model/Music.kt:6]: na nakon inkaro
bar aksse daghighan
ba intafavot ke `LocalMusic` faghat baraye pass dadan be fragment dige estefade mishe na chize dige
`Music` estelahan contract e ma hast baraye laye haye "model" mesle Repository ya ViewModel

File: app/src/main/java/com/worldsnas/starplayer/model/Repository.kt (hunk 10 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/model/Repository.kt]: 100% behtar az ine
nazare digaran ham beporsid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt (hunk 16 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: bashe

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt (hunk 3 above)
Comment (ghost) [reply app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt]: chon inja kolan list hame ahang hai ke harooz server migere miyare ro neshon mide
vali linke haye dge hast ke maslan byad search bezanim dakhele ahang on moghe parametre vorodi migre.
ke harche zood tar in feature o ezafe mikonam

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt (hunk 4 above)
Comment (ghost) [reply app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt:7]: apiInterface khobe?

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt (hunk 3 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt]: Lotfan ba paging piade konid

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt (hunk 4 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt:7]: Na
Mitonid az digaran ham soal konid

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 19]
```diff
@@ -127,8 +127,8 @@ class MusicsListFragment : Fragment() {
         }
     }
 
-    private fun musicListener(localMusic: LocalMusic) {
-        val bundle = bundleOf(ConstValues.BUNDLE_KEY_MUSIC_INFO to localMusic)
+    private fun musicListener(music: Music) {
+        val bundle = bundleOf(ConstValues.BUNDLE_KEY_MUSIC_INFO to music)
 
         findNavController()
             .navigate(R.id.action_musicsListFragment_to_playerFragment, bundle)
```
Comment (hosseiniSeyRo) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: Az safe args estefade kind behtare

File: app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt (hunk 14 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/RepositoryImpl.kt]: doroste inja behesh eshare shode 
https://stackoverflow.com/questions/58083309/coroutines-and-retrofit-best-way-to-handle-errors

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 19 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: chera , dalil khasi dare?
in kare badi hast?

File: app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt (hunk 3 above)
Comment (ghost) [reply app/src/main/java/com/worldsnas/starplayer/api/WebServiceApi.kt]: > Lotfan ba paging piade konid

chashm

File: app/src/main/java/com/worldsnas/starplayer/api/MusicResponse.kt (hunk 1 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/api/MusicResponse.kt]: lotan hatman az `@SerializeName()` estefade konin

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+    private val webServiceApi: WebServiceApi
+) : MusicRepository {
+
+    override suspend fun getApiData(): List<LocalMusic> {
+        val localMusicList: MutableList<LocalMusic> = ArrayList<LocalMusic>().toMutableList()
+        return withContext(Dispatchers.IO) {
+            val musicList = webServiceApi.getMusics(1,10)
+
+            if (musicList.isSuccessful && musicList.body()?.isNotEmpty()!!) {
+                musicList.body()?.forEach {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: mitonid ba `.map` bezanid ke kheili behtar mishe

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+                            it.name,
+                            it.artist,
+                            "album",
+                            "genre",
+                            it.musicLink
+                        )
+                    localMusicList += music
+                }
+            }
+            return@withContext localMusicList
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: niazi be `return@withContext` nist

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+        }
+    }
+
+    override suspend fun getLocalData(): List<LocalMusic> =
+        withContext(Dispatchers.IO) {
+            return@withContext localMusicProvider.getAllMusic()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: hamchenin
`return@withContext`

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
still the LocalMusic is wrong and has to be fixed

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 19 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: amniat bishtari mide nesbat be code

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt (hunk 21 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: nemizare mige hatman bayad bashe

Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: bedone in kar nemikone

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt (hunk 20 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: manzooreton parametr haye endpoint hast?

File: app/src/main/java/com/worldsnas/starplayer/api/MusicResponse.kt [hunk 23]
```diff
@@ -0,0 +1,14 @@
+package com.worldsnas.starplayer.api
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class MusicResponse(
+    @Json(name = "id") val id: Long,
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/api/MusicResponse.kt:8]: afarin

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+
+class MusicRepositoryImpl @Inject constructor(
+    private val localMusicProvider: LocalMusicProvider,
+    private val webServiceApi: WebServiceApi
+) : MusicRepository {
+
+    override suspend fun getApiData(): List<MusicRepoModel> {
+        val musicRepoModelList: MutableList<MusicRepoModel> = ArrayList<MusicRepoModel>().toMutableList()
+        return withContext(Dispatchers.IO) {
+            val musicList = webServiceApi.getMusics(1,10)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: gharar bod in data az biron biad

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class MusicRepositoryImpl @Inject constructor(
+    private val localMusicProvider: LocalMusicProvider,
+    private val webServiceApi: WebServiceApi
+) : MusicRepository {
+
+    override suspend fun getApiData(): List<MusicRepoModel> {
+        val musicRepoModelList: MutableList<MusicRepoModel> = ArrayList<MusicRepoModel>().toMutableList()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: `.toMutableList()` niazi nist

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt (hunk 20 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: `forEach()` -> `.map()`

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+                            it.name,
+                            it.artist,
+                            "album",
+                            "genre",
+                            it.musicLink
+                        )
+                    musicRepoModelList += music
+                }
+            }
+            return@withContext musicRepoModelList
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: goftam niazi be in return nist

File: app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt [hunk 27]
```diff
@@ -71,8 +72,8 @@ class PlayerFragment : Fragment() {
     }
 
     private fun setMusicArguments() {
-        music = arguments?.getParcelable(ConstValues.BUNDLE_KEY_MUSIC_INFO)
-            ?: throw Throwable("Did not Receive Anything ")
+        val args: PlayerFragmentArgs by navArgs()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: ino fek konam beshe be onvane `property` tarifesh kard. nemishe?

File: app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt [hunk 28]
```diff
@@ -25,7 +26,7 @@ class PlayerFragment : Fragment() {
     private lateinit var playerComponent: PlayerComponent
     private var exoPlayer: SimpleExoPlayer? = null
     private var viewBinding: FragmentPlayerBinding? = null
-    private lateinit var music: LocalMusic
+    private lateinit var music: Music
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: in ham `Music` khali nabashe behtare
`MusicArg` chetore?

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 29]
```diff
@@ -5,29 +5,31 @@ import androidx.lifecycle.LiveData
...
-import com.worldsnas.starplayer.model.LocalMusicProviderImpl
-import com.worldsnas.starplayer.model.Music
+import com.worldsnas.starplayer.model.MusicRepoModel
+import com.worldsnas.starplayer.model.MusicRepositoryImpl
 import kotlinx.coroutines.*
 import javax.inject.Inject
 
-class MusicListViewModel @Inject constructor(private val localMusicProvider: LocalMusicProvider) :
+class MusicListViewModel @Inject constructor(
+    private val repositoryImpl: MusicRepositoryImpl
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: lotfan interface ro inject konin

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
looks perfect
just final touches

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt (hunk 29 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: doroste havasam nabode

File: app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt (hunk 28 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: AAlie

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt (hunk 26 above)
Comment (neptu39701) [reply app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt:33]: Agha khata migire

Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: done

File: app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt (hunk 27 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: navArgs lazy hast

Comment (neptu39701) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt:75]: ببرم گلوبال کنم بعد پاس بدم به مدل مقدار رو؟

Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: koja bezaramesh?

Comment (neptu39701) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt:75]: Global Bezaram?

Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: in kar ro kardam az share nullable khalas besham

Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: in kar ham mishe kard
https://paste.ubuntu.ir/rilv

Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt]: to badane class nemishe bashe?

File: app/src/main/java/com/worldsnas/starplayer/di/modules/MusicProviderModule.kt [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+import com.worldsnas.starplayer.api.WebServiceApi
+import com.worldsnas.starplayer.model.LocalMusicProvider
+import com.worldsnas.starplayer.model.LocalMusicProviderImpl
+import com.worldsnas.starplayer.model.MusicRepository
+import com.worldsnas.starplayer.model.MusicRepositoryImpl
+import dagger.Module
+import dagger.Provides
+
+@Module
+object MusicProviderModule {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/di/modules/MusicProviderModule.kt:13]: vaghti @Inject mizarim balaye constrcutor dige niazi be in file nist

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+                    val music =
+                        MusicRepoModel(
+                            it.id.toInt(),
+                            it.name,
+                            it.artist,
+                            "album",
+                            "genre",
+                            it.musicLink
+                        )
+                    musicRepoModelList += music
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: aziatam nakon dige!
.map mizani dige niazi be in kar nist!!!

File: app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt [hunk 32]
```diff
@@ -25,7 +26,8 @@ class PlayerFragment : Fragment() {
     private lateinit var playerComponent: PlayerComponent
     private var exoPlayer: SimpleExoPlayer? = null
     private var viewBinding: FragmentPlayerBinding? = null
-    private lateinit var music: LocalMusic
+    private lateinit var musicArgs: Music
+    private val args: PlayerFragmentArgs by navArgs()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt:30]: ahan afarin

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
Not much further

File: app/src/main/java/com/worldsnas/starplayer/di/modules/MusicProviderModule.kt (hunk 30 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/di/modules/MusicProviderModule.kt:13]: vaghti pak mikonm khata mide ke provide nashodan

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 33]
```diff
@@ -0,0 +1,39 @@
...
+import javax.inject.Inject
+
+class MusicRepositoryImpl @Inject constructor(
+    private val localMusicProvider: LocalMusicProvider,
+    private val webServiceApi: WebServiceApi
+) : MusicRepository {
+
+    override suspend fun getApiData(page: Int, count: Int): List<MusicRepoModel> =
+        withContext(Dispatchers.IO) {
+            lateinit var musicRepoModelList: List<MusicRepoModel>
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: dige che niazi be in variable hast aslan?

File: app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt [hunk 34]
```diff
@@ -0,0 +1,39 @@
...
+            val apiMusicList = webServiceApi.getMusics(page, count)
+
+            if (apiMusicList.isSuccessful && apiMusicList.body()?.isNotEmpty()!!) {
+                musicRepoModelList = apiMusicList.body()?.map {
+
+                    val music = MusicRepoModel(
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt]: be in variable e `music` ham niazi nist

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
brief and to the point

**Review by helloseyedjafari — APPROVED** [review]
Good job buddy
you did well

**Pull request merged**